app/src/main/java/org/schabi/newpipe/local/feed/FeedFragment.java

```diff
@@ -159,220 +193,173 @@ public void readFrom(@NonNull Queue<Object> savedObjects) throws Exception {
     @Override
     public void startLoading(boolean forceLoad) {
         if (DEBUG) Log.d(TAG, "startLoading() called with: forceLoad = [" + forceLoad + "]");
-        if (subscriptionObserver != null) subscriptionObserver.dispose();
 
-        if (allItemsLoaded.get()) {
-            if (infoListAdapter.getItemsList().size() == 0) {
-                showEmptyState();
-            } else {
-                showListFooter(false);
-                hideLoading();
-            }
-
-            isLoading.set(false);
-            return;
-        }
+        if (!hasStartedLoading.get() || forceLoad) {
+            if (subscriptionObserver != null) subscriptionObserver.dispose();
 
-        isLoading.set(true);
-        showLoading();
-        showListFooter(true);
-        subscriptionObserver = subscriptionService.getSubscription()
+            setLoadingState(true);
+            subscriptionObserver = subscriptionService.getSubscription()
                 .onErrorReturnItem(Collections.emptyList())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(this::handleResult, this::onError);
+                .observeOn(Schedulers.io())
```

Comment:
Are you sure this is ok? `handleResult` and `onError` would be run on a thread different from the main one, and since they do gui stuff this could potentially create problems. I could be wrong, though ;-)

Reply:
Since all the UI related operations are executed inside the callbacks of `Handler::post`, they will eventually be executed in the main thread. I think this is okay.

Reply:
Oh, ok. Thank you for the clarification ;-)